app/services/mosaic_service.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+# This class provides mosaic manifest uris, and calls out to the
+# MosaicGenerator as needed
+class MosaicService
+  class Error < StandardError; end
+  attr_reader :resource
+  # @param resource [RasterResource]
+  def initialize(resource:)
+    @resource = resource.decorate
+  end
+
+  # calculate the path for the non-fingerprinted version of the file
+  def path
+    raise Error if raster_file_sets.empty?
+    mosaic_path = Valkyrie::Storage::Disk::BucketedStorage.new(base_path: base_path).generate(resource: resource, original_filename: "mosaic.json", file: nil).to_s
+    return mosaic_path if storage_adapter.find_by(id: mosaic_file_id)
+  rescue Valkyrie::StorageAdapter::FileNotFound
+    build_node if MosaicGenerator.new(output_path: tmp_file.path, raster_paths: raster_paths).run
```

Comment:
Suggestion: Call `tmp_file.close` after you're done, otherwise we risk leaving that file handle open.